### ISSUE TITLE
Crash in RegisterInstanceLoadHandlerOnce 

### ISSUE THREAD

**Issue by FlashHit** [VeniceUnleashed/VeniceUnleashed#661]
Doing `ResourceManager:RegisterInstanceLoadHandlerOnce` within `ResourceManager:RegisterInstanceLoadHandlerOnce` and it's the same instance.

```lua
class 'TEST'

function TEST:__init()
	ResourceManager:RegisterInstanceLoadHandler(Guid('993445AF-2476-11E0-834E-C984E80F7234'), Guid('5FE2571D-D0AD-CF75-3CB6-43A43AFC0E8B'), self, self.OnInstanceLoaded)
end

function TEST:OnInstanceLoaded(p_Instance)
	-- I want that this gets called again the next time this instance gets loaded
	ResourceManager:RegisterInstanceLoadHandler(Guid('993445AF-2476-11E0-834E-C984E80F7234'), Guid('5FE2571D-D0AD-CF75-3CB6-43A43AFC0E8B'), self, self.OnInstanceLoaded)
end

if g_TEST == nil then
	g_TEST = TEST()
end

```
Here as a mod:
[RegisterInstanceCrash.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/6819037/RegisterInstanceCrash.zip)

**Comment by OrfeasZ** [comment]
What map / mode can this be repro'd in?

**Comment by FlashHit** [comment]
I think every map, but I did it in:
`XP5_003 ConquestLarge0 2`

Because the instance that I load is: `UI/Flow/Screen/IngameMenuMP`
So that should be loaded in every map/ mode.

**Comment by OrfeasZ** [comment]
Fixed in `17655`.

**Comment by FlashHit** [comment]
This fixes the crash but it doesn't really work like I hoped for it:

This callback gets fired only once. And I hoped it would fire again after a map switch.
```lua
class 'TEST'

function TEST:__init()
	ResourceManager:RegisterInstanceLoadHandlerOnce(Guid('993445AF-2476-11E0-834E-C984E80F7234'), Guid('5FE2571D-D0AD-CF75-3CB6-43A43AFC0E8B'), self, self.OnInstanceLoaded)
end

function TEST:OnInstanceLoaded(p_Instance)
	-- I want that this gets called again the next time this instance gets loaded
	print("OnInstanceLoaded")
	ResourceManager:RegisterInstanceLoadHandlerOnce(Guid('993445AF-2476-11E0-834E-C984E80F7234'), Guid('5FE2571D-D0AD-CF75-3CB6-43A43AFC0E8B'), self, self.OnInstanceLoaded)
end

if g_TEST == nil then
	g_TEST = TEST()
end
```
Note: I switched this to `RegisterInstanceLoadHandlerOnce`

**Comment by OrfeasZ** [comment]
Are you sure this instance is reloaded on map switch? If you look for it using the `Partition:Loaded` event does it show up? If it does, then open a different issue for this.

**Comment by FlashHit** [comment]
alright, and yes I am sure.
Will create a new issue rn.
